File: algorithms/week10-dinamyc-programming/B.cpp
#include <iostream>
#include <vector>
using namespace std;

const int MOD = 1e9 + 7;

int solve(int n, int x, const vector<int>& coins) {
    vector<int> count(x + 1, 0);
    count[0] = 1;

    for (int i = 1; i <= x; ++i) {
        for (int coin : coins) {
            if (i - coin >= 0) {
                count[i] = (count[i] + count[i - coin]) % MOD;
            }
        }
    }

    return count[x];
}

int main() {
    int n, x;
    cin >> n >> x;
    vector<int> coins(n);
    for (int i = 0; i < n; ++i) {
        cin >> coins[i];
    }

    cout << solve(n, x, coins) << endl;
    return 0;
}

File: algorithms/week10-dinamyc-programming/B.py
n, x = map(int, input().split())
coins = list(map(int, input().split()))

count = [0] * (x+1)
count[0] = 1
m = (10**9) + 7

for i in range(1, x+1):
    for c in coins:
        if (i-c) >= 0:
            count[i] += count[i-c]
            count[i] %= m
            
print(count[x])